File: scripts/run-legal-shadow-pack-validation.mjs
import { createServer } from "vite";
import { fileURLToPath } from "node:url";
import { dirname, resolve } from "node:path";
const root = resolve(dirname(fileURLToPath(import.meta.url)), "..");
const server = await createServer({ root, resolve: { alias: { "@": resolve(root, "src") } }, server: { middlewareMode: true }, appType: "custom" });
try { await server.ssrLoadModule("/scripts/run-legal-shadow-pack-validation.ts"); }
finally { await server.close(); }

File: scripts/run-legal-shadow-pack-validation.py
#!/usr/bin/env python3
"""Bundle and run the TypeScript shadow validation against the real compiler."""
import pathlib,subprocess
root=pathlib.Path(__file__).resolve().parents[1]
raise SystemExit(subprocess.run(['node','scripts/run-legal-shadow-pack-validation.mjs'],cwd=root).returncode)
